### PR TITLE
Changes random strings to use constants

### PR DESCRIPTION
**Pull request by iamkirkbater**
Also edits the constants to use CONSTANT case for clarity.

Still needs to be tested locally, I'm working to set up my environment now.

### CODE REVIEW

**Comment by iamkirkbater** [comment]
This has been tested locally and is now ready for review.

**Comment by clcollins** [comment]
Looks good to me, but I think it suffers from the same bug @fahlmant was fixing in #259, so might need to rebase on that when it's merged?

**Comment by iamkirkbater** [comment]
I originally did that in my first attempt, but ended up trying to generalize too far.  Actually, though, I have an idea that I'll run with and get back to you on this.

**Comment by iamkirkbater** [comment]
@fahlmant I originally tried putting those consts in the `utils` package, but that ended up creating a circular dependency within the `v1alpha1` package when I tried using them there as well.  So I just put them in the `v1alpha1` package and then used them from there.

Let me know if you have any concerns with that, otherwise this should be good if you think it is.

**Comment by jharrington22** [comment]
@iamkirkbater general feedback here, with the naming of these variables its important to remember that the style of your variables is also related to syntax, eg the first letter of a variable denotes whether or not its exported from the package. As for style Go is always camel case besides that first letter of course.

Sharing the same variable across types is probably not a good idea as we want to keep types defined and contained to themselves ie. changing v1alpha1.READY would effect more than one types eg `v1alpha1.AccountConditionType.AccountReady` and `v1alpha1.ClaimStatus.ClaimStatusReady` this wouldn't be expected. When we define a type we expect that type to be the source of truth.

File: pkg/controller/account/account_controller.go [hunk 1]
```diff
@@ -462,7 +462,7 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	// Account init for both BYOC and Non-BYOC
-	if (currentAcctInstance.Spec.BYOC && currentAcctInstance.Status.State != "Ready") || ((currentAcctInstance.Status.State == string(awsv1alpha1.AccountCreating)) && (currentAcctInstance.Status.Claimed == false)) {
+	if (currentAcctInstance.Spec.BYOC && currentAcctInstance.Status.State != AccountReady) || ((currentAcctInstance.Status.State == string(awsv1alpha1.AccountCreating)) && (currentAcctInstance.Status.Claimed == false)) {
```
Comment (fahlmant) [inline pkg/controller/account/account_controller.go]: I think `string(awsv1alpha1.AccountCreating)` can also be changed to `AccountCreating`

**Pull request merged**